examples/multi_processor/main.cc

```diff
@@ -34,7 +34,7 @@ void initTracer()
   auto exporter2 = std::unique_ptr<trace_sdk::SpanExporter>(new InMemorySpanExporter());
 
   // fetch the exporter for dumping data later
-  memory_span_exporter = dynamic_cast<InMemorySpanExporter *>(exporter2.get());
+  memory_span_exporter = static_cast<InMemorySpanExporter *>(exporter2.get());
```

Comment:
Could this `dynamic_cast` be removed by assigning `new InMemorySpanExporter()` to `memory_span_exporter` at first?

Reply:
The constructor of processor only accept the base class.It's better practice to alwarys wrapper raw pointer with smart pointer because it can be destroyed if there is any exception later. But in this case, it's simple between we call `new operator` and construct the `unique_ptr`. I think it's fine to assign `memory_span_exporter` first.